core/src/main/java/org/apache/iceberg/util/ContentFileUtil.java

```diff
@@ -49,7 +50,21 @@ public static <F extends ContentFile<K>, K> K copy(
     }
   }
 
+  /**
+   * @deprecated since 1.7.0, will be removed in 2.0.0; use {@link
+   *     #referencedDataFileLocation(DeleteFile)} instead.
+   */
+  @Deprecated
   public static CharSequence referencedDataFile(DeleteFile deleteFile) {
+    return referencedLocation(deleteFile);
+  }
+
+  public static String referencedDataFileLocation(DeleteFile deleteFile) {
+    CharBuffer location = referencedLocation(deleteFile);
+    return location != null ? location.toString() : null;
+  }
+
+  private static <T> T referencedLocation(DeleteFile deleteFile) {
```

Comment:
Is there value in a new method? Why not simply call `referencedDataFile` in the new method?

```
public static String referencedDataFileLocation(DeleteFile deleteFile) {
  CharSequence location = referencedDataFile(deleteFile);
  return location != null ? location.toString() : null;
}
```

Reply:
Good point, no not really. Just how I refactored it, in my head I was just isolating the extraction of the location to a separate method. Since there's no real  value add, we may as well make the change more minimal.